tests/test_adversary.py

```diff
@@ -135,13 +135,39 @@ def test_hidden_params_after_forward(input_data, target_data, perturbation):
 
     output_data = adversary(input=input_data, target=target_data, model=model, sequence=sequence)
 
-    # Adversarial perturbation will not have parameter even after forward is called.
+    # Adversarial perturbation will have one parameter after forward is called.
     params = [p for p in adversary.parameters()]
-    assert len(params) == 0
+    assert len(params) == 1
 
-    # Adversarial perturbation should not have any state dict items being exported to the model checkpoint.
+    # Adversarial perturbation should have one state dict item being exported to the model checkpoint.
     state_dict = adversary.state_dict()
-    assert len(state_dict) == 0
+    assert len(state_dict) == 1
+
+
+def test_loading_perturbation_from_state_dict():
+    initializer = Mock()
+    composer = mart.attack.composer.Additive()
+    projector = Mock()
+
+    perturber = Perturber(initializer=initializer, projector=projector)
+
+    gain = Mock()
+    enforcer = Mock()
+    attacker = Mock(max_epochs=0, limit_train_batches=1, fit_loop=Mock(max_epochs=0))
+
+    adversary = Adversary(
+        perturber=perturber,
+        composer=composer,
+        optimizer=None,
+        gain=gain,
+        enforcer=enforcer,
+        attacker=attacker,
+    )
+
+    # We should be able to load arbitrary state_dict, because Adversary ignores state_dict.
+    # We want this behavior for Adversary because model checkpoints may include perturbation in state_dict
+    # that is not loadable before initialization of perturbation.
+    adversary.load_state_dict({"perturbation": None})
```

Comment:
Can you set perturbation to something (e.g., `{"perturbation": tensor([1, 2, 3]}`) and check that perturber.perturbation is not that value? That seems like the behavior to check?

Reply:
I also don't quite understand the comment. Why would it fail? I guess this is because perturbation is `None` so it doesn't find a parameter to load. Maybe that means we should change perturbation to be an `UninitializedParameter` instead of None...

Reply:
No, we can not load a tensor from `state_dict` to `UninitializedParameter`.

Reply:
You are right. The test fails without the patch in this PR, because perturbation is `None`.

Reply:
What does "can not load" mean? This seems to be fine:

```python
>>> import torch
>>> class UPModule(torch.nn.Module):
...   def __init__(self):
...     super().__init__()
...     self.param = torch.nn.UninitializedParameter()
...
>>> upm = UPModule()
>>> upm.load_state_dict({"param": torch.Tensor([1.])})
<All keys matched successfully>
>>> upm.param
<UninitializedParameter>
```

But looking at the code closer it's probably better to use `None` since we also use `ParameterList` and I don't know how that would work with `UninitializedParameter`.